tree/dataframe/inc/ROOT/RDF/RInterface.hxx

```diff
@@ -2051,26 +2054,7 @@ private:
 
       fLoopManager->Book(actionPtr.get());
 
-      // create new RDF
-      ::TDirectory::TContext ctxt;
-      // Now we mimic a constructor for the RDataFrame. We cannot invoke it here
-      // since this would introduce a cyclic headers dependency.
-
-      // Keep these two statements separated to work-around an ABI incompatibility
-      // between clang (and thus cling) and gcc in the way std::forward is handled.
-      // See https://sft.its.cern.ch/jira/browse/ROOT-9236 for more detail.
-      auto rlm_ptr = std::make_shared<RLoopManager>(nullptr, validCols);
-      auto snapshotRDF = std::make_shared<RInterface<RLoopManager>>(rlm_ptr);
-      auto chain = std::make_shared<TChain>(fullTreename.c_str());
-      chain->Add(std::string(filename).c_str());
-      snapshotRDF->fProxiedPtr->SetTree(chain);
-      auto snapshotRDFResPtr = MakeResultPtr(snapshotRDF, *fLoopManager, std::move(actionPtr));
-
-      if (!options.fLazy) {
-         *snapshotRDFResPtr;
-      }
-
-      return snapshotRDFResPtr;
+      return RDFInternal::CreateSnaphotRDF(validCols, fullTreename, std::string(filename), options.fLazy, *fLoopManager, std::move(actionPtr));
```

Comment:
taking a `string_view` would avoid this string creation here

Reply:
But will need a string in the function implementation :)